### ISSUE TITLE
Support sampled concurrent verification in MLLM MTP scheduling

### ISSUE THREAD

**Issue by Thump604** [waybarrios/vllm-mlx#659]
## Capability gap

The MLLM MTP scheduler uses greedy argmax equality for verification and
previously bypassed sampled or concurrent requests. That prevents native MTP
from serving normal request sampling tuples through continuous batching.

## Exact local repro

Run a Qwen 3.6 35B MLLM server with continuous batching, MTP, and normal
sampling requests:

```sh
python -m runtime_patches serve <qwen-model> --mllm --continuous-batching \
  --enable-mtp --mtp-num-draft-tokens 1
```

Two concurrent requests using `temperature`, `top_p`, `top_k`, and `min_p`
completed with visible content and `finish_reason=stop`. Scheduler metadata
reported 6 draft attempts, 5 accepted drafts, 1 exact fallback rejection, and
0 errors.

## Upstream code path compared

Compared local and `main` at `0dd115769ef1196a715b96b181353edacd2a4f69`:

- `vllm_mlx/mllm_batch_generator.py::install_mtp_mllm`

## Expected behavior

MTP verification uses each request's post-filter sampling distribution. State
that crosses scheduler turns is keyed by request UID, not the batch position.
When one row rejects in a concurrent batch, the scheduler replays the primary
token for every row rather than using an invalid per-row cache rollback.

## Minimal patch

- Add exact sampled speculative acceptance and residual sampling.
- Preserve request-local verified state across UID reordering.
- Hold newly queued work for the one verified-state transition.
- Keep MTP disabled for logits-processor requests because their transformed
  distributions are not available to the draft verifier.

## Not claimed

This request does not add multi-token MTP recurrence, constrained-output
support under MTP, or tool-call MTP support.

**Comment by Thump604** [comment]
Resolved by #662, which is merged. Closing as completed.
